src/vm/docs/contracts.rs

```diff
@@ -85,20 +85,70 @@ is expired, or if there's never been such a stacker, then returns none."),
+        ("name-import", "Import name to a revealed namespace. Each imported name is given both an owner and some off-chain state."),
+        ("namespace-ready", "Launches the namespace and makes it available to the public. Once a namespace is launched, anyone can register a name in it if they pay the appropriate amount of cryptocurrency."),
+        ("name-preorder", "Preorders a name by telling all BNS nodes the salted hash of the BNS name. It pays the registration fee to the namespace owner's designated address."),
+        ("name-register", "Reveals the salt and the name to all BNS nodes, and assigns the name an initial public key hash and zone file hash."),
+        ("name-update", "Changes the name's zone file hash. You would send one of these transactions if you wanted to change the name's zone file contents. For example, you would do this if you want to deploy your own Gaia hub and want other people to read from it."),
+        ("name-transfer", "Changes the name's public key hash. You would send one of these transactions if you wanted to:
+* Change your private key
+* Send the name to someone else
+
+When transferring a name, you have the option to also clear the name's zone file hash (i.e. set it to null). This is useful for when you send the name to someone else, so the recipient's name does not resolve to your zone file."),
+        ("name-revoke", "Makes a name unresolvable. The BNS consensus rules stipulate that once a name is revoked, no one can change its public key hash or its zone file hash.  The name's zone file hash is set to null to prevent it from resolving. You should only do this if your private key is compromised, or if you want to render your name unusable for whatever reason."),
+        ("name-renewal", "Depending in the namespace rules, a name can expire. For example, names in the .id namespace expire after 2 years. You need to send a name reveal every so often to keep your name.
+
+You will pay the registration cost of your name to the namespace's designated burn address when you renew it.
+When a name expires, it enters a month-long \"grace period\" (5000 blocks). 
+
+It will stop resolving in the grace period, and all of the above operations will cease to be honored by the BNS consensus rules.
+You may, however, send a NAME_RENEWAL during this grace period to preserve your name.
+If your name is in a namespace where names do not expire, then you never need to use this transaction."),
+        ("get-namespace-price", "Get the price for a namespace."),
+        ("get-name-price", "Get the price for a name."),
+        ("can-namespace-be-registered", "Returns true if the provided namespace is available."),
+        ("is-name-lease-expired", "Return true if the provided name lease is expired."),
+        ("can-name-be-registered", "Returns true if the provided name can be registered."),
```

Comment:
What are the conditions? Otherwise the comment is not very helpful.

Reply:
which command explicitly? If you know the conditions, please let me know. I'm not an expert on the contract - I really just want to help with _some_ documentation